src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/ExceptionHandlingUtil.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.InvalidCaseDataException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CallbackException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.ProcessResult;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+final class ExceptionHandlingUtil {
+    private ExceptionHandlingUtil() {
+        //
+    }
+
+    static CallbackException handleGenericException(Exception exception, String msg) {
+        // log happens individually to cover update/ccd cases
+        return new CallbackException(msg, exception);
+    }
+
+    static ProcessResult handleInvalidCaseDataException(InvalidCaseDataException exception, String msg) {
```

Comment:
Also not sure why this is needed. Is it getting repeated at different places ?

Reply:
This whole class is to fix a quality gate failure due to code duplication (as mentioned in the comment to the PR)

Reply:
Could you please post link to sonar qube regarding the quality gate issue as I am not sure it looks like util class is used at only one place ?